app/assets/javascripts/channels/desks.js

```diff
@@ -10,6 +10,31 @@ App.cable = ActionCable.createConsumer();
 
 App.messages = App.cable.subscriptions.create('DesksChannel', {
   received: function(data) {
-    $("#desk_state_" + data.id)[0].innerHTML = data.occupied;
+
+      OCCUPIED_CLASS = 'btn-outline-danger';
+      NOT_OCCUPIED_CLASS = 'btn-outline-success';
+
+      // Get desk state button
+      var target = $("#desk_state_" + data.id)[0];
+
+      if(!target)
```

Comment:
This may be too late to `return`. At this point, there would already be an index out of bounds error. If we want to account for invalid ids we'll have to do it before indexing the query selector's response. ie. `[0]`

Reply:
I believe in javascript this will return null, so I think this is valid.

In my browsers console I just did:

(Where 'desk_state_test' doesn't exist)

> $("#desk_state_test")[0]
null

Reply:
My apologies, I actually get undefined!